src/libraries/Microsoft.Extensions.Caching.Abstractions/src/MemoryCacheExtensions.cs

```diff
@@ -75,8 +76,9 @@ public static bool TryGetValue<TItem>(this IMemoryCache cache, object key, out T
         /// <returns>The value that was set.</returns>
         public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value)
         {
-            using ICacheEntry entry = cache.CreateEntry(key);
+            ICacheEntry entry = cache.CreateEntry(key);
             entry.Value = value;
+            entry.Dispose();
```

Comment:
How is this better than the `using`?

Reply:
It avoids the `try-finally` overhead which is not needed in this case.